---
title: "Visualizing_genome"
author: "Paulyna Magana"
date: "19/05/2022"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi = 1000,
                      echo = TRUE,
                      cache = TRUE)
```

### Libraries

```{r, message=FALSE, warning=FALSE}
library(Gviz)
library(rtracklayer)
library(GenomicRanges)
library(biomaRt)
```

```{r tracknames, warning=FALSE}

session <- browserSession()
genome(session) <- "hg38"
trackNames(session)

```

```{r example, warning=FALSE, cache=TRUE, fig.width=8, fig.height=4}
gtrack <- GenomeAxisTrack() #to add a genomic axis to the plot
ideoTrack <- IdeogramTrack(genome = "hg38", chromosome = "chr6")# ideogram

#plot
# showbandId - chromosome bands in the ideogram
# turn off the explicit plotting of the chromosome name by setting the showId display parameter to FALSE.
plotTracks(ideoTrack, from = 85e6, to = 129e6, showId = FALSE, 
           showBandId = TRUE, cex.bands = 0.5)
```

# SLC22A1

```{r SLC22A1, warning=FALSE, cache=TRUE, fig.width=8, fig.height=4}

###Simple annotation tracks:
fromA1 <- 160115734
toA1 <- 160160664



##Gene model type tracks:

ncbi_ref <- UcscTrack(genome="hg38", chromosome="chr6", track="NCBI RefSeq", 
                    from=fromA1, to=toA1, trackType="GeneRegionTrack", 
                    rstarts="exonStarts", rends="exonEnds", gene="name", 
                    symbol="name", transcript="name", strand="strand", 
                    fill="#767676",stacking="squish", name="NCBI RefSeq", 
                    showId=TRUE, geneSymbol=TRUE)


biomTrack <- BiomartGeneRegionTrack(genome="hg38", chromosome="chr6",
                                    start=fromA1, end=toA1, name="Ensembl")


###Data tracks



###Other tracks and plotting: 

axTrack <- GenomeAxisTrack(labelPos="below")
idxTrack <- IdeogramTrack(genome="hg38", chromosome="chr6")

##PLOTTING
plotTracks(list(idxTrack, axTrack, ncbi_ref), from=fromA1, toA1, showTitle=TRUE, col.line=NULL,
           collapseTranscripts="longest", transcriptAnnotation = "symbol")


```

## highlight CRISPR-Cas9 region

```{r SLC22A1 high, warning=FALSE, cache=TRUE, fig.width=8, fig.height=4}
ht <- HighlightTrack(trackList = list(axTrack, ncbi_ref),
                     start = c(160136234, 160133991), width = 10,
                     chromosome = 6)

lim <- c(fromA1, toA1)

plotTracks(list(idxTrack, ht), from = lim[1], to = lim[2])

```

## luciferase regions

```{r}
library(tidyverse)

luciferase_plasmids <- read_csv("./luciferase_plasmids.csv")

as.data.frame(luciferase_plasmids)

luciferase_plasmids$start <- as.integer(luciferase_plasmids$start)
luciferase_plasmids$end <- as.integer(luciferase_plasmids$end)
luciferase_plasmids$width <- as.integer(luciferase_plasmids$width)
luciferase_plasmids$chromosome <- as.factor(luciferase_plasmids$chromosome)
luciferase_plasmids$gene <- as.factor(luciferase_plasmids$gene)


head(luciferase_plasmids)
```

```{r SLC22A1 LUCIFERASE, warning=FALSE, cache=TRUE, fig.width=12, fig.height=4}
grtrack <- GeneRegionTrack(luciferase_plasmids, genome = "hg38",
                           chromosome = "chr6",
                           name = "Luciferase regions")

plotTracks(list(idxTrack,axTrack, ncbi_ref, grtrack), from = 160118000, to = 160126000, transcriptAnnotation="gene")

```

# SLC22A5

```{r SLC22A5, warning=FALSE, cache=TRUE, fig.width=12, fig.height=4}

###Simple annotation tracks:
fromA5 <- 132366800
toA5 <- 132396017


##Gene model type tracks:

ncbi_ref <- UcscTrack(genome="hg38", chromosome="chr5", track="NCBI RefSeq", 
                    from=fromA5, to=toA5, trackType="GeneRegionTrack", 
                    rstarts="exonStarts", rends="exonEnds", gene="name", 
                    symbol="name", transcript="name", strand="strand", 
                    fill="#767676",stacking="squish", name="NCBI RefSeq", 
                    showId=TRUE, geneSymbol=TRUE)


biomTrack <- BiomartGeneRegionTrack(genome="hg38", chromosome="chr5",
                                    start=fromA5, end=toA5, name="Ensembl")


###Data tracks



###Other tracks and plotting: 

axTrack <- GenomeAxisTrack(labelPos="below")
idxTrack <- IdeogramTrack(genome="hg38", chromosome="chr5")

##PLOTTING
plotTracks(list(idxTrack, axTrack, ncbi_ref), from=fromA5, to=toA5, showTitle=TRUE, col.line=NULL,
           collapseTranscripts="longest", transcriptAnnotation = "symbol")

```

## highlight CRISPR-Cas9 region

```{r SLC22A5 high, warning=FALSE, cache=TRUE, fig.width=12, fig.height=4}

ht <- HighlightTrack(trackList = list(axTrack, ncbi_ref),
                     start = c(132384160, 132378383), width = 10,
                     chromosome = 5)

lim <- c(fromA5, toA5)

plotTracks(list(idxTrack, ht), from = lim[1], to = lim[2])


```

## luciferase regions


```{r SLC22A5 LUCIFERASE, warning=FALSE, cache=TRUE, fig.width=12, fig.height=6}
grtrack <- GeneRegionTrack(luciferase_plasmids, genome = "hg38",
                           chromosome = "chr5",
                           name = "Luciferase regions")

plotTracks(list(idxTrack,axTrack, ncbi_ref, grtrack), from = fromA5, to = 132389321, transcriptAnnotation="gene")

```


# SLC22A4

```{r SLC22A4, warning=FALSE, cache=TRUE, fig.width=14, fig.height=4}

###Simple annotation tracks:
fromA4 <- 132290928
toA4 <- 132345343


##Gene model type tracks:

ncbi_ref <- UcscTrack(genome="hg38", chromosome="chr5", track="NCBI RefSeq", 
                    from=fromA4, to=toA4, trackType="GeneRegionTrack", 
                    rstarts="exonStarts", rends="exonEnds", gene="name", 
                    symbol="name", transcript="name", strand="strand", 
                    fill="#767676",stacking="squish", name="NCBI RefSeq", 
                    showId=TRUE, geneSymbol=TRUE)


biomTrack <- BiomartGeneRegionTrack(genome="hg38", chromosome="chr5",
                                    start=fromA4, end=toA4, name="Ensembl")


###Data tracks



###Other tracks and plotting: 

axTrack <- GenomeAxisTrack(labelPos="below")
idxTrack <- IdeogramTrack(genome="hg38", chromosome="chr5")

##PLOTTING
plotTracks(list(idxTrack, axTrack, ncbi_ref), from=fromA4, to=toA4, showTitle=TRUE, col.line=NULL,
           collapseTranscripts="longest", transcriptAnnotation = "symbol")

```

## highlight CRISPR-Cas9 region

```{r SLC22A4 high, warning=FALSE, cache=TRUE, fig.width=16, fig.height=4}

ht <- HighlightTrack(trackList = list(axTrack, ncbi_ref),
                     start = c(132313626, 132312204), width = 7,
                     chromosome = 5)

lim <- c(fromA4, toA4)

plotTracks(list(idxTrack, ht), from = lim[1], to = lim[2])


```

## luciferase regions


```{r SLC22A4 LUCIFERASE, warning=FALSE, cache=TRUE, fig.width=12, fig.height=4}
grtrack <- GeneRegionTrack(luciferase_plasmids, genome = "hg38",
                           chromosome = "chr5",
                           name = "Luciferase regions")

plotTracks(list(idxTrack,axTrack, ncbi_ref, grtrack), from = 132293519, to = 132295119, transcriptAnnotation="gene")

```




## Session Info

```{r sessioninfo, echo=FALSE}
sessionInfo()
```
