---
title: "Visualizing_genome"
author: "Paulyna Magana"
date: "19/05/2022"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries

```{r, message=FALSE, warning=FALSE}
library(Gviz)
library(GenomicRanges)
```


```{r}

data(cpgIslands)
class(cpgIslands)

chr <- as.character(unique(seqnames(cpgIslands)))
gen <- genome(cpgIslands) #plot the single CpG islands annotation track.

```

```{r}
atrack <- AnnotationTrack(cpgIslands, name= "CpG") #plot cpgislands
gtrack <- GenomeAxisTrack() #to add a genomic axis to the plot
ideoTrack <- IdeogramTrack(genome = "hg19", chromosome = "chr6")# ideogram
```


```{r plot}
#plot
# showbandId - chromosome bands in the ideogram
# turn off the explicit plotting of the chromosome name by setting the showId display parameter to FALSE.
plotTracks(ideoTrack, from = 85e6, to = 129e6, showId = FALSE, 
           showBandId = TRUE, cex.bands = 0.5)
```


```{r, cache=TRUE}

###Simple annotation tracks:

from <- 160120744
to <- 160128417
cpgIslands <- UcscTrack(genome="hg38", chromosome="chr6",
                        track="cpgIslandExt", from=from, to=to,
                        trackType="AnnotationTrack", start="chromStart", end="chromEnd",
                        id="name", shape="box", fill="#006400", name="CpG Islands")


snpLocations <- UcscTrack(genome="hg38", chromosome="chr6",
                          track="All SNPs(144)", from=from, to=to, trackType="AnnotationTrack",
                          start="chromStart", end="chromEnd", id="name", feature="func",
                          strand="strand", shape="box", stacking="dense", fill="black",
                          name="SNPs")

##Gene model type tracks:

knownGenes <- UcscTrack(genome="hg38", chromosome="chr6",
                        track="knownGene", from=from, to=to, trackType="GeneRegionTrack",
                        rstarts="exonStarts", rends="exonEnds", gene="name", symbol="name",
                        transcript="name", strand="strand", fill="#8282d2", name="UCSC Genes")


refGenes <- UcscTrack(genome="hg38", chromosome="chr6",
                      track="xenoRefGene", from=from, to=to,
                      trackType="GeneRegionTrack", rstarts="exonStarts",
                      rends="exonEnds", gene="name", symbol="name2",
                      transcript="name", strand="strand", fill="#8282d2",
                      stacking="dense", name="Other RefSeq")

ensGenes <- UcscTrack(genome="hg38", chromosome="chr6",
                      track="All GENCODE V25", from=from, to=to, trackType="GeneRegionTrack",
                      rstarts="exonStarts", rends="exonEnds", gene="name",
                      symbol="name2", transcript="name", strand="strand",
                      fill="#960000", name="Ensembl Genes")


###Data tracks

conservation <- UcscTrack(genome="hg38", chromosome="chr6",
                          track="Conservation", table="phastCons100way", from=from, to=to,
                          trackType="DataTrack", start="start", end="end", data="score",
                          type="hist", window="auto", col.histogram="darkblue",
                          fill.histogram="darkblue", ylim=c(-3.7, 4), name="Conservation")

gcContent <- UcscTrack(genome="hg38", chromosome="chr6",
                       track="GC Percent", table="gc5Base", from=from, to=to,
                       trackType="DataTrack", start="start", end="end", data="score",
                       type="hist", window=-1, windowSize=1500, fill.histogram="black",
                       col.histogram="black", ylim=c(30, 70), name="GC Percent")


###Other tracks and plotting: 


axTrack <- GenomeAxisTrack()
idxTrack <- IdeogramTrack(genome="hg38", chromosome="chr6")
plotTracks(list(idxTrack, axTrack, knownGenes, refGenes,
                ensGenes, cpgIslands, gcContent, conservation,
                snpLocations), from=from, to=to, showTitle=FALSE)




```


```{r}

biomTrack <- BiomartGeneRegionTrack(genome="hg38", chromosome="chr6",
                                    start=from, end=to, name="Ensembl")
plotTracks(biomTrack)

```


```{r}
plotTracks(biomTrack, col.line=NULL) 
```



## R Markdown


```{r pressure, echo=FALSE}
sessionInfo()
```
